sdk/inc/azure/core/az_json.h

```diff
@@ -792,6 +792,43 @@ AZ_NODISCARD az_result az_json_reader_next_token(az_json_reader* ref_json_reader
  */
 AZ_NODISCARD az_result az_json_reader_skip_children(az_json_reader* ref_json_reader);
 
+/**
+ * @brief Gets the JSON token's string after unescaping it, if required.
+ *
+ * @param[in] json_string The #az_span that contains the string to be unnescaped.
+ * @param destination Pointer to the buffer that will contain the oputput.
+ * @param[in] destination_max_size The maximum available space within the buffer referred to by
+ * \p destination.
+ * @param[out] out_string_length __[nullable]__ Contains the number of bytes written to the
+ * \p destination which denote the length of the unescaped string. If `NULL` is passed, the
+ * \p parameter is ignored.
+ *
+ * @return An #az_result value indicating the result of the operation.
+ * @retval #AZ_OK The string is returned.
+ * @retval #AZ_ERROR_NOT_ENOUGH_SPACE \p destination does not have enough size.
+ */
+AZ_NODISCARD az_result az_json_string_unescape(
+    az_span json_string,
+    char* destination,
+    int32_t destination_max_size,
+    int32_t* out_string_length);
+
+/**
+ * @brief Gets the JSON token's string after unescaping it in place, if required.
+ *
+ * @param[in] json_string The #az_span that contains the string to be unnescaped in place.
+ * @param[out] out_string_length __[nullable]__ Contains the number of valid bytes in the span
+ * \p which denote the length of the unescaped string. If `NULL` is passed, the
+ * \p parameter is ignored.
+ *
+ * @return An #az_result value indicating the result of the operation.
+ * @retval #AZ_OK The string is returned.
+ * @retval #AZ_ERROR_NOT_ENOUGH_SPACE \p destination does not have enough size.
+ */
+
+AZ_NODISCARD az_result
+az_json_string_unescape_in_place(az_span* json_string, int32_t* out_string_length);
```

Comment:
1. @CIPop  liked this API 
2. @ahsonkhan  please elaborate, the API works so i'm not sure where your'going with it, we also update the size in the span to the new size.
3. Apparently they would need more the in place rather than out of place due to memory constraints. 
Removing this API saves 10s -100s of bytes in the process code when loaded in memory, then use KBs in the data due to having to duplicate buffers. that does not seem like an overall loss in memory for the sake of doctrine.